backend/apid/middlewares/authorization.go

```diff
@@ -44,3 +49,41 @@ func (a Authorization) Then(next http.Handler) http.Handler {
 		next.ServeHTTP(w, r.WithContext(ctx))
 	})
 }
+
+// BasicAuthorization performs basic authorization for entity creation via the agent websocket.
+func BasicAuthorization(next http.Handler, store store.Store) http.Handler {
+	return http.HandlerFunc(func(w http.ResponseWriter, r *http.Request) {
+		namespace := r.Header.Get(transport.HeaderKeyNamespace)
+		ctx := r.Context()
+		ctx = context.WithValue(ctx, corev2.NamespaceKey, namespace)
+
+		user, err := store.GetUser(ctx, r.Header.Get(transport.HeaderKeyUser))
```

Comment:
I don't believe we should do this, considering the agent user could be within an authentication provider. Instead, we should rely on the claims injected into the context by [`middlewares.BasicAuthentication`](https://github.com/sensu/sensu-go/blob/c80ddead34075326a45ae9d3f878304c6c43ce53/backend/apid/middlewares/authentication.go#L84) and fetch those like this: https://github.com/sensu/sensu-go/blob/a82a17edaf9f7455c5d86cc2a63bbe5bfc152835/backend/apid/middlewares/authorization_attributes.go#L120

Reply:
Good point! Fixing this also made me realize I had the authorization and authentication middlewares in the wrong order. 2 🐦1 stone!